addons/website_slides/static/src/js/activity.js

```diff
@@ -42,4 +42,178 @@ function applyInclude(Activity) {
+             * Refuse the access for the current `Course`.
+             */
+            async refuseAccess() {
+                await this.async(() => this.env.services.rpc({
+                    model: 'slide.channel',
+                    method: 'action_refuse_access',
+                    args: [this.id, this.partnerId],
+                }));
+                if (this.activity) {
+                    this.activity.delete();
+                }
+                this.delete();
+            }
+
+            //----------------------------------------------------------------------
+            // Private
+            //----------------------------------------------------------------------
+
+            /**
+             * @override
+             */
+            static _createRecordLocalId(data) {
+                return `${this.modelName}_${data.id}`;
+            }
+
+        }
+
+        Slide.fields = {
+            activity: one2one('mail.activity', {
+                inverse: 'slide',
+            }),
+            id: attr(),
+            partnerId: attr(),
+        };
+
+        Slide.modelName = 'slides.slide';
+
+        return Slide;
+    }
+
+    registerNewModel('slides.slide', factory);
+
+});
+
+odoo.define('website_slides/static/src/js/slide.js', function (require) {
+    'use strict';
+
+    const components = {
+        Activity: require('mail/static/src/components/activity/activity.js'),
+    };
+    const useStore = require('mail/static/src/component_hooks/use_store/use_store.js');
+
+    class Slide extends owl.Component {
+
+        /**
+         * @override
+         */
+        constructor(...args) {
+            super(...args);
+            useStore(props => {
+                const slide = this.env.models['slides.slide'].get(props.slideLocalId);
+                return {
+                    slide: slide ? slide.__state : undefined,
+                };
+            });
+        }
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * @returns {slides.slide}
+         */
+        get slide() {
+            return this.env.models['slides.slide'].get(this.props.slideLocalId);
+        }
+
+        //--------------------------------------------------------------------------
+        // Handlers
+        //--------------------------------------------------------------------------
+
+        /**
+         * @private
+         */
+        async _onGrantAccess(ev) {
+            await this.slide.grantAccess();
+            this.trigger('reload');
+        }
+        /**
+         * @private
+         */
+        async _onRefuseAccess(ev) {
+            await this.slide.refuseAccess();
+            this.trigger('reload');
+        }
+    }
+
+    Object.assign(Slide, {
+        props: {
+            slideLocalId: String,
+        },
+        template: 'slides.Slide',
```

Comment:
Rename: template name should contain module name and have name of component class after dot `website_slides.Slide`